pulp_2to3_migration/app/migration.py

```diff
@@ -177,17 +179,20 @@
         message='Migrating distributors to Pulp 3', code='migrating.distributors', total=0
     )
     with ProgressReport(**progress_data) as pb:
-        pulp2distributors_qs = Pulp2Distributor.objects.filter(
-            pulp3_distribution=None,
-            pulp3_publication=None,
-            not_in_plan=False)
-        pb.total = pulp2distributors_qs.count()
-        pb.save()
+        for plugin in plan.get_plugin_plans():
+            distributor_types = list(plugin.migrator.distributor_migrators.keys())
+            pulp2distributors_qs = Pulp2Distributor.objects.filter(
+                pulp3_distribution=None,
+                pulp3_publication=None,
+                not_in_plan=False,
+                pulp2_type_id__in=distributor_types
+            )
+            pb.total = pulp2distributors_qs.count()
+            pb.save()
 
-        # gather all needed plugin distributor migrators
-        distributor_migrators = {}
+            # gather all needed plugin distributor migrators
+            distributor_migrators = {}
 
-        for plugin in plan.get_plugin_plans():
             distributor_migrators.update(**plugin.migrator.distributor_migrators)
```

Comment:
since we process distributors by plugin type now, i believe we can just assign directly `distributor_migrators = plugin.migrator.distributor_migrators`, no need to update the dict